tasks/human_eval_031.rs

```diff
@@ -9,10 +9,77 @@ use vstd::prelude::*;
 
 verus! {
 
-// TODO: Put your solution (the specification, implementation, and proof) to the task here
+// specification
+pub closed spec fn prime_helper(n: nat, k: nat) -> bool
+    recommends
+        2 <= k <= n,
+    decreases n - k,
+{
+    if k < 2 || k > n {
+        false  //arbitrarily chosen return value, should have 2 <= k <= n
+
+    } else if n >= k >= n - 1 {
+        true
+    } else if n % k == 0 {
+        false
+    } else {
+        prime_helper(n, k + 1)
+    }
+}
+
+pub open spec fn is_prime(n: nat) -> bool {
```

Comment:
This spec is nice in that it's executable, but `prime_helper` seems slightly convoluted to include in a trusted spec.  What if instead the spec was this?
```
n > 1 && forall |i| 1 < i < n ==> n % i != 0
```
That might be easier for a reviewer to decide matches the expected definition of primality.

Reply:
I ran into the same issue with this spec which Amar ran into last week: https://github.com/verus-lang/verus/issues/1277. Not sure if I should wait for this to be resolved (would it be something I could learn to resolve myself?) or keep my original spec.

Reply:
Resolving that issue may be rather involved; let's talk about it on Wednesday.  For now, I'd be inclined to use the cleaner spec and comment out the sanity check for now.

Reply:
done! commented out the sanity check and verified an updated implementation using the cleaner spec.